### PR TITLE
Add new recipe ocilib/4.7.6

### PR DESCRIPTION
**Pull request by inPhraZ**
### Summary
Added new recipe:  **ocilib/4.7.6**

#### Motivation

This PR will address issue https://github.com/conan-io/conan-center-index/issues/23867

#### Details

Reopening #23954 with squashed commits.

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

File: recipes/ocilib/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        with chdir(self, self.source_folder):
+            autotools = Autotools(self)
+            autotools.configure()
+            autotools.make()
+
+    def package(self):
+        autotools = Autotools(self)
+        autotools.install()
+
+        copy(self, "COPYING", self.source_folder, os.path.join(self.package_folder, "licenses"))
```
Comment (uilianries) [inline recipes/ocilib/all/conanfile.py:69]:
```suggestion
        copy(self, "COPYING", self.source_folder, os.path.join(self.package_folder, "licenses"))
        copy(self, "LICENSE", self.source_folder, os.path.join(self.package_folder, "licenses"))
```

Need to copy license file too: https://github.com/vrogier/ocilib/blob/v4.7.6/LICENSE

File: recipes/ocilib/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.15)
+project(test_package LANGUAGES CXX)
+
+set(CMAKE_CXX_STANDARD 11)
+
+find_package(ocilib REQUIRED CONFIG)
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} PRIVATE ocilib::ocilib)
```
Comment (uilianries) [inline recipes/ocilib/all/test_package/CMakeLists.txt:9]:
```suggestion

find_package(ocilib REQUIRED CONFIG)

add_executable(${PROJECT_NAME} test_package.cpp)
target_link_libraries(${PROJECT_NAME} PRIVATE ocilib::ocilib)
target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_11)
```

Prefer CMake target features

**Review by uilianries — COMMENTED** [review]
Fair enough supporting only Linux for now, it can be improved in a newer PR. 

The project offers support for Windows: https://github.com/vrogier/ocilib/tree/master/proj/dll, but it will require extra effort.

**Review by uilianries — APPROVED** [review]
LGTM. Built locally on Linux for checking, the recipe looks good, no extra files installed, no requirements listed. 

Thank you for you PR!

**Review by AbrilRBS — APPROVED** [review]
Thanks!

**Pull request merged**